packages/zoltan2/core/src/input/Zoltan2_BasicIdentifierAdapter.hpp

```diff
@@ -116,31 +122,89 @@ template <typename User>
   BasicIdentifierAdapter(lno_t numIds, const gno_t *idPtr):
       numIds_(numIds), idList_(idPtr), weights_() {}
 
+  /*! \brief Constructor
+   *  \param ids should point to a View of identifiers.
+   *  \param weights  a list of pointers to arrays of weights.
+   *      The number of weights per identifier is assumed to be
+   *      \c weights.extent(0).
+   *  \param weightStrides  a list of strides for the \c weights.
+   *     The weight for weight index \c n for \c ids[k] should be
+   *     found at <tt>weights[n][weightStrides[n] * k]</tt>.
+   *     If \c weightStrides.size() is zero, it is assumed all strides are one.
+   *
+   *  The values pointed to the arguments must remain valid for the
+   *  lifetime of this Adapter.
+   */
+  BasicIdentifierAdapter(
+    Kokkos::View<gno_t*, device_t> &ids,
+    Kokkos::View<scalar_t**, device_t> &weights);
```

Comment:
We should probably use type aliases (we created) in all occurences of `Kokkos::View`. This is helpful if we decide to change the type, then we only have to change it in one place - type alias definition.

Reply:
Unfortunately, due to time constraints, this will have to be done a future contract